### ISSUE TITLE
Is ‘Using GCC with MinGW' section in the vscode document kind of outdated?

### ISSUE THREAD

**Issue by tasty0tomato** [microsoft/vscode-docs#4092]
mingw-x64的下载方式MingW-W64-builds（即文档中提供的下载方式）最后更新时间为2018年，并且其提供的gcc版本最新仅为8.1.0，与现在gcc官方网站维护的8.3 9.2 10.0版本落后较大时间了。相反的，也是我认为的替代品，MSYS2提供着持续更新的gcc与mingw版本，还拥有多个镜像源。经过我的尝试，除了安装gcc过程稍有不同，它无疑是MingW-W64-builds在现如今的替代品。另外是与Cygwin的比较，从某种程度上，二者较为相似，但是Cygwin拥有一个致命的”缺陷“，不提供卸载程序，所以更显得MSYS2更合适现如今的Using GCC with MinGW。

综上所述，我认为Using GCC with MinGW应该转为采用MSYS2并且更新文档相关部分

This is my own translation, and there may be problems.
The mingw-x64 download method 'MingW-W64-builds' (i.e., the download method available in the documentation) was last updated in 2018, and the gcc version it provides is 8.1 7.3 6.4 5.4 , which is much behind the 8.3 9.2 10.0 version now maintained by the GCC official website. On the contrary, and as a substitute, MSYS2 offers a continuously updated version of gcc and mingw, as well as multiple mirror sources. After my attempt, it is undoubtedly a nice replacement for MingW-W64-builds today, except that the installation gcc process is slightly different. The other is a comparison with Cygwin, which is somewhat similar, but Cygwin has a fatal "disadvantage" that does not provide an uninstall executable file, so it is more appropriate for MSYS2 to be the current 'Using GCC with MinGW'.

In summary, I think Using GCC with MinGW should move to MSYS2 and update the relevant parts of the documentation

**Comment by tasty0tomato** [comment]
by the way, maybe wsl will be a nice choice.

**Comment by TylerMSFT** [comment]
Hi @tasty0tomato ,

Thank you for the idea. I'll look into it, but rewriting it around MSYS2, if we decide to do that, would need to come after several other projects.
 
Regarding the point that using WSL would be a good idea, we have a topic already that does just that :-)  See https://code.visualstudio.com/docs/cpp/config-wsl

Closing since I've added this to my list of work to consider.

**Comment by tasty0tomato** [comment]
@TylerMSFT  May I ask what you choose if you need to work with mingw? I am kind of confused.

**Comment by TylerMSFT** [comment]
I'm not sure what you mean by "what to choose if you work with mingw"? Are you asking what to download?

**Comment by tasty0tomato** [comment]
@TylerMSFT  Yes,

**Comment by TylerMSFT** [comment]
@tasty0tomato , if you click the link in step 3, "Install Mingw-w64 via the SourceForge website. Click Mingw-w64 to download the Windows Mingw-w64 installer", it will automatically download the file you need to install. In Edge, for example, you'll see the file downloading in the tray at the bottom.

**Comment by MightBeHarmless** [comment]
Working on a project that is built using a tool chain that looks like:
- MSYS2
- packages including git, meson, ninja and a bunch of libraries (mingw-w64 where appropriate)

I can work with the project using the MinGW 64 command prompt quite well.
Visual Studio Code was brought in as my preferred editor & IDE, but it's been troublesome to configure, due to the available information not being relevant to some of the tools that I'm using not covered by existing documentation or other factors.

In fact I'm having specific problems with Code not recognizing or functioning with the integrated Git based SCM. 

Would also really like to update our project's source with functioning Build & Debugging through Code.

Anyhow, Just writing this note because most web searches point me to this documentation repository, though I have not been able to solve my issues with this documentation in its current state. Should I get lucky to find out how to overcome the blocking problems with Git integration and setting up Build or Debugging, I'd like to be able to contribute to documentation here that would benefit many other users.
